### PR TITLE
Media lib front fixes

### PR DESCRIPTION
**Pull request by virginieky**
#### Description of what you did:
Some front-end fixes

### CODE REVIEW

File: packages/strapi-helper-plugin/lib/src/utils/getFilterType.js [hunk 1]
```diff
@@ -102,6 +102,21 @@ const getFilterType = type => {
           value: '_ncontains',
         },
       ];
+    case 'size':
```
Comment (soupette) [inline packages/strapi-helper-plugin/lib/src/utils/getFilterType.js]: I am not sure this belong in this file as size is not an attribute type but more something you need to add in order to make your code works you should find another way to do it in the file where this is used.

File: packages/strapi-plugin-upload/admin/src/components/EditForm/index.js [hunk 2]
```diff
@@ -79,7 +79,9 @@ const EditForm = forwardRef(
     useEffect(() => {
       if (isImg || isVideo) {
         if (prefixedFileURL) {
-          setSrc(prefixedFileURL);
+          // Adding performance.now forces the browser no to cache the img
+          // https://stackoverflow.com/questions/126772/how-to-force-a-web-browser-not-to-cache-images
+          setSrc(`${prefixedFileURL}?${performance.now()}`);
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/components/EditForm/index.js]: I don't think we need to add this here as the file will be fetched on every re reader furthermore since the file is fetched when the modal is opened there's no cache. Can you explain why you added it?

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js [hunk 3]
```diff
@@ -235,14 +232,22 @@ const HomePage = () => {
...
+    setIsPopupOpen(false);
 
-    dispatch({
-      type: 'CLEAR_DATA_TO_DELETE',
-    });
+    lockAppWithOverlay();
 
-    setIsPopupOpen(false);
-    fetchListData();
+    await Promise.all(dataToDelete.map(item => deleteMedia(item.id)))
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js]: You can do this in a try catch instead of a promise

File: packages/strapi-plugin-upload/admin/src/components/EditForm/index.js [hunk 4]
```diff
@@ -79,7 +80,9 @@ const EditForm = forwardRef(
     useEffect(() => {
       if (isImg || isVideo) {
         if (prefixedFileURL) {
-          setSrc(prefixedFileURL);
+          // Adding performance.now forces the browser no to cache the img
+          // https://stackoverflow.com/questions/126772/how-to-force-a-web-browser-not-to-cache-images
+          setSrc(`${prefixedFileURL}?${cacheRef.current}`);
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/components/EditForm/index.js]: I think you can remove it completely and just use the url.

File: packages/strapi-plugin-upload/admin/src/components/FiltersPicker/FiltersCard/utils/filtersForm.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ const filtersForm = {
     defaultValue: moment(),
   },
   size: {
-    type: 'size',
+    type: 'integer',
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/components/FiltersPicker/FiltersCard/utils/filtersForm.js]: Good 👍

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js (hunk 3 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js]:
```
try {
  await Promise.all()

  dispatch({
     type: 'CLEAR_DATA_TO_DELETE',
  });

  fetchListData();
} catch(error) {
 // Silent
} finally {
 strapi.unlockApp();
}

File: packages/strapi-plugin-upload/admin/src/components/SortPicker/index.js [hunk 6]
```diff
@@ -10,12 +10,12 @@ import Picker from '../Picker';
 
 const SortPicker = ({ onChange, value }) => {
   const orders = {
-    created_at_asc: 'created_at:ASC',
     created_at_desc: 'created_at:DESC',
+    created_at_asc: 'created_at:ASC',
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/components/SortPicker/index.js]: I don't understand these changes I prefer when it was sorted. Is there any reason for modifying the order?

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js [hunk 7]
```diff
@@ -47,22 +47,33 @@ const HomePage = () => {
...
-
-  const { data, dataCount, dataToDelete, isLoading } = reducerState.toJS();
+  const { data, dataCount, dataToDelete, isLoading, searchParams } = reducerState.toJS();
   const pluginName = formatMessage({ id: getTrad('plugin.name') });
   const paramsKeys = ['_limit', '_start', '_q', '_sort'];
   const debouncedSearch = useDebounce(searchValue, 300);
 
   useEffect(() => {
     handleChangeParams({ target: { name: '_q', value: searchValue } });
+
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js]: `debouncedSearch` should be use instead of `searchValue` so you can remove the eslint-lint override.

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js [hunk 8]
```diff
@@ -47,22 +47,33 @@ const HomePage = () => {
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   useEffect(() => {
     fetchListData();
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [search]);
+  }, [JSON.stringify(searchParams)]);
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js]: You should use the search here.

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js [hunk 9]
```diff
@@ -40,6 +46,24 @@ const reducer = (state, action) => {
...
+          .updateIn(['searchParams', '_start'], () => 0);
+      }
+
+      return state.updateIn(['searchParams', name], () => value);
+    }
+    case 'SET_PARAMS': {
+      const { params } = action;
+
+      return state.update('searchParams', searchParams => {
+        return searchParams.mergeDeep(fromJS(params));
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js]: You should keep the data immutable here. Maybe you can use a `reduce` to merge the new params. Let me know if you are struggling here.

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js [hunk 10]
```diff
@@ -40,6 +46,24 @@ const reducer = (state, action) => {
 
       return state.removeIn(['dataToDelete', index]);
     }
+    case 'SET_PARAM': {
+      const { name, value } = action;
+
+      if (name === '_limit') {
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js]: I don't see this case in your tests.

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/tests/reducer.test.js [hunk 11]
```diff
@@ -320,4 +320,50 @@ describe('Upload | containers | HomePage | reducer', () => {
 
     expect(reducer(state, action)).toEqual(expectedState);
   });
+
+  it('should update searchParams with given params', () => {
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/tests/reducer.test.js]: Can you wrap your test in a `describe` with the name of the corresponding action so it is easier to identify what your are testing?

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import { isEmpty, toString } from 'lodash';
+
+const generateStringFromParams = query => {
+  let params = '';
+
+  Object.keys(query)
+    .filter(key => !isEmpty(toString(query[key])))
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js]: Why do you need a `toString` here?

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    .forEach(key => {
+      const value = query[key];
+
+      if (key === 'filters') {
+        value.forEach(item => {
+          if (item.name.includes('mime') && item.value === 'file') {
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js]: `if (item.name === 'mime' && item.value === 'file')`  you don't need the `includes` here.

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+            const revertedKey = item.filter.includes('_ncontains')
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js]: Same here about the includes

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+              ? 'mime_contains'
+              : 'mime_ncontains';
+            const filterValue = `${revertedKey}=image&${revertedKey}=video`;
+
+            params += `&${filterValue}`;
+          } else {
+            const name = item.filter === '=' ? item.name : `${item.name}${item.filter}`;
+            params += `&${name}=${item.value}`;
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/generateStringFromParams.js]: can you add an extra line here?

File: packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/tests/generateStringFromParams.test.js [hunk 16]
```diff
@@ -0,0 +1,75 @@
+import generateStringFromParams from '../generateStringFromParams';
+
+describe('MEDIA LIBRARY | containers | HomePage | utils', () => {
```
Comment (soupette) [inline packages/strapi-plugin-upload/admin/src/containers/HomePage/utils/tests/generateStringFromParams.test.js]: Can you add a test where the search (`_q`) is empty?

File: packages/strapi-plugin-upload/admin/src/components/SortPicker/index.js (hunk 6 above)
Comment (virginieky) [reply packages/strapi-plugin-upload/admin/src/components/SortPicker/index.js]: The UI

File: packages/strapi-plugin-upload/admin/src/components/ListWrapper/index.js [hunk 17]
```diff
@@ -1,7 +1,7 @@
 import styled from 'styled-components';
 
 const ListWrapper = styled.div`
-  margin-top: 2px;
+  margin-top: 20px;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/ListWrapper/index.js]: Don't forget to check if this value match in the content manager modal. If not, you can just create another wrapper for the plugin case and leave the old value by default.

File: packages/strapi-plugin-upload/admin/src/components/UploadList/Container.js [hunk 18]
```diff
@@ -3,7 +3,7 @@ import ContainerFluid from '../ContainerFluid';
 
 const Container = styled(ContainerFluid)`
   margin-bottom: 4px;
-  padding-top: 15px;
+  padding-top: 20px;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/UploadList/Container.js]: Same here, as the UploadList is used in the content manager modal, check if it works with this new value.

File: packages/strapi-plugin-upload/admin/src/components/ListWrapper/index.js (hunk 17 above)
Comment (virginieky) [reply packages/strapi-plugin-upload/admin/src/components/ListWrapper/index.js]: you're right
